File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i=0,j=height.size()-1;
        int max_vol=0;
        while(j>i){
            int area=min({height[i],height[j]})*(j-i);
            max_vol=max({area,max_vol});
            if(height[i]<height[j]){
                i++;
            }else{
                j--;
            }
        }
        return max_vol;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int max_water = 0;
        while (left < right) {
            int h = Math.min(height[left], height[right]);
            int width = right - left;
            max_water = Math.max(max_water, h * width);
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return max_water;
    }
}